xdsl/dialects/test.py

```diff
@@ -46,6 +48,10 @@ class TestTermOp(TestOp):
 
     name = "test.termop"
 
+    res: VarOpResult = var_result_def()
```

Comment:
Does it make sense for a terminator to have results?
This is something we used in some of the tests. That's why my earlier try failed ([CI run](https://github.com/xdslproject/xdsl/actions/runs/5611853160/jobs/10268770422))

Reply:
My one question was also about SSAValue scoping which I was not 100% sure about anymore. If SSAValues are block-scoped, this does not make sense as the terminator is the last operation in the block. If they are region-scoped (which from a quick google search, they seem to be in MLIR), I guess you could build IR that uses it.

And then the follow up question is: Will there ever be a use for it? Who would define a terminator that defines an SSAValue?

I guess we can leave it as-is. Better test more behavior than what people actually use. But, it was something that made me raise my eyebrow, so I decided to comment on it :sweat_smile:

Reply:
> I guess you could build IR that uses it.

Yeah, true, I missed that point.
But my doubts remain on the utility of this, and when in Rome, I'd rather do what MLIR does (or does not do in this case :-)

Reply:
Technically, a terminator can have regions, and even results.
Should they? I don't think so, but *maybe* someone one day will have a use case for it, so we can keep it as MLIR does.

Reply:
I added regions as well. Let's keep it like this...